libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -294,13 +301,21 @@ void NavEKF3_core::setAidingMode()
             // reset the vertical position state to faster recover from baro errors experienced during touchdown
             stateStruct.position.z = -meaHgtAtTakeOff;
         } else if (PV_AidingMode == AID_RELATIVE) {
-            // We have commenced aiding, but GPS usage has been prohibited so use optical flow only
-            GCS_MAVLINK::send_statustext_all(MAV_SEVERITY_INFO, "EKF3 IMU%u is using optical flow",(unsigned)imu_index);
+            // We are doing relative position navigation where velocity errors are constrained, but position drift will occur
+            if (readyToUseOptFlow()) {
+                GCS_MAVLINK::send_statustext_all(MAV_SEVERITY_INFO, "EKF3 IMU%u is using optical flow",(unsigned)imu_index);
+                // Reset time stamps
+                flowValidMeaTime_ms = imuSampleTime_ms;
+                prevFlowFuseTime_ms = imuSampleTime_ms;
+            } else if (readyToUseBodyOdm()) {
+                GCS_MAVLINK::send_statustext_all(MAV_SEVERITY_INFO, "EKF3 IMU%u is using odometry",(unsigned)imu_index);
```

Comment:
This is a notification bug. Both sensors types can be used together. This notification is unfortunately tied to the transition from constant position to relative position mode and therefore only the first sensor type that passes checks gets a notification message.

I will be reworking the notification logic.